### ISSUE TITLE
Line 144 of code

### ISSUE THREAD

**Issue by mengxiangming** [sajaddarabi/BiRealNet#1]
Hi, thanks for sharing the code and I have one question about the line 144 of Bi-Real-Net where self.tanh1 is defined but not used anywhere else. Is there something missing? Thanks.

**Comment by sajaddarabi** [comment]
Hi, 

self.tanh1 is really a placeholder that could be used as an activation after the first bn1, though I believe in the original birealnet they don't do so.

I'll try to push the complete training code this weekend or the following week once I have some headroom :)

**Comment by mengxiangming** [comment]
Hi Sajad:

Thanks so much for your reply. 

I noted that in your code, the activation seems to be added in the self.layer1 where the self._make_layer uses tanh as the first operation. So I think they may be the same as the original one. 


Recently I only achieved Top1 accuracy 13% in 7 epochs using your network, which is quite slow compared to full precision case (10-20 in 1 epoch). What is your Top1 acc  vs. epochs? Is it also such slow? And what is your final Top1 acc? 


Looking forward to your complete code since I am doing some comparisons. Is it possible to send me first by email since I am in a hurry. 

Thanks very very much. 

Xiangming

**Comment by sajaddarabi** [comment]
Hi Xiang Ming,

Yes,  it should converge properly if you use the correct settings.

I've added pretrained weights (which should probably save you a week..), and accuracy curves in README.md.

The top-1 accuracies I get for the fp model at epoch 120 for bireal18 and bireal34 are 67.928% and 70.246% respectively.

As for the code, I currently don't have access to my laptop. I'll try to push it end of this week or early next week.

Hope this helps :)
Sajad

**Comment by mengxiangming** [comment]
Hi Sajad,

Thanks so much for your kindness. I see it in your repo the results are
quite good. I will have a try with it though actrually I hope to train
Bi-Rreal Net from scratch:)

Thanks!
Best regards,

Xiangming

Sajad Darabi <notifications@github.com> 于 2019年9月18日周三 04:11写道：
